src/OpenTelemetry/Resources/Resource.cs

```diff
@@ -101,28 +101,33 @@ public Resource Merge(Resource other)
                 sanitizedKey = attribute.Key;
             }
 
-            object sanitizedValue;
-            if (!IsValidValue(attribute.Value))
-            {
-                OpenTelemetrySdkEventSource.Log.InvalidArgument("Create resource", "attribute value", "Attribute value should be a non-null string, long, bool or double.");
-                sanitizedValue = string.Empty;
-            }
-            else
-            {
-                sanitizedValue = attribute.Value;
-            }
+            object sanitizedValue = SanitizeValue(attribute.Value);
 
             return new KeyValuePair<string, object>(sanitizedKey, sanitizedValue);
         }
 
-        private static bool IsValidValue(object value)
+        private static object SanitizeValue(object value)
         {
-            if (value != null && (value is string || value is bool || value is long || value is double))
+            if (value != null)
             {
-                return true;
+                if (value is string || value is bool || value is long || value is double)
+                {
+                    return value;
+                }
+
+                if (value is int)
+                {
+                    return System.Convert.ToInt64(value);
+                }
+
+                if (value is float)
+                {
+                    return System.Convert.ToDouble(value);
+                }
             }
 
-            return false;
+            OpenTelemetrySdkEventSource.Log.InvalidArgument("Create resource", "attribute value", "Attribute value should be a non-null primitive or homogeneous array of primitives.");
+            return string.Empty;
```

Comment:
@cijothomas @reyang Currently if you give Resource an invalid value it writes an empty string (throws away the value). Should this be a throw instead? I feel like since Resource is created during startup it might be OK to throw here? Let the user know they've made a mistake so they can correct it.

Reply:
Given its at startup, its fair to throw and let user correct it.

Reply:
Done. Using System.ArgumentException, is this an appropriate use of a base exception class?